crates/dbsp/src/trace/spine_async/mod.rs

```diff
@@ -195,6 +195,24 @@ where
         self.slots[level].loose_batches.push_back(batch);
     }
 
+    fn should_apply_backpressure(&self) -> bool {
+        const HIGH_THRESHOLD: usize = 128;
+        self.slots
+            .iter()
+            .map(|s| s.loose_batches.len())
+            .sum::<usize>()
+            >= HIGH_THRESHOLD
+    }
+
+    fn should_relieve_backpressure(&self) -> bool {
```

Comment:
It took me a while to understand the naming. I know it's longer but maybe `should_release_backpressure_waiters` would be a better name.